components/feature/search/src/main/java/mozilla/components/feature/search/widget/BaseVoiceSearchActivity.kt

```diff
@@ -0,0 +1,132 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.search.widget
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.speech.RecognizerIntent
+import androidx.activity.result.ActivityResult
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.annotation.VisibleForTesting
+import androidx.appcompat.app.AppCompatActivity
+import java.util.Locale
+
+/**
+ * Launches voice recognition then uses it to start a new web search.
+ */
+abstract class BaseVoiceSearchActivity : AppCompatActivity() {
+
+    /**
+     * Holds the intent that initially started this activity
+     * so that it can persist through the speech activity.
+     */
+    private var previousIntent: Intent? = null
+
+    @VisibleForTesting
+    private var activityResultLauncher: ActivityResultLauncher<Intent>? = null
```

Comment:
Why do we create this result launcher in `onCreate` to only use it later on in the same callback method?

Also, you use the `@VisibleForTesting`, but I don't see any tests that use it.

Reply:
Done

Reply:
It's still unclear to me why we need to hold this in a var in this fashion. :)